src/misc.rs

```diff
@@ -47,6 +50,190 @@ where
         .unwrap_or_else(|| Vec::new())
 }
 
+/// Given as input `elements`, an array of field elements
+/// \\(\rho_0, \dots, \rho_{n-1}\\)
+/// compute the tensor product
+/// \\( \otimes_j (1, \rho_j )\\)
+pub fn tensor<F: Field>(elements: &[F]) -> Vec<F> {
+    assert!(!elements.is_empty());
```

Comment:
why not just put an if statement / match and return () if the tensor vector is empty?

Reply:
fixed

Reply:
fixed